main.go

```diff
@@ -78,6 +85,30 @@ func main() {
 		regArr = append(regArr, re)
 	}
 
+	for u := 0; u < loopCount; u++ {
```

Comment:
Do you need this if no files are specified?

Reply:
no but if there are no files specified the program will either print an error or just stuck in an endless loop (grep does the same thing)

Reply:
No, it would read from stdin.

Reply:
You're already doing the same. Avoid the loop in this case.

Reply:
Oke but I dont get the problem

Reply:
Non-clean code is the problem.

Reply:
so i have to check if there are no files specified? and if there are none i have to just end the program or print and error message. The question is how do I check if there are no files specified??

Reply:
C'mon, read your own code. **You already check** whether there are any files specified or not and set `loopCount = 1`.

Reply:
I only check if there are 0 flag.Args() -> i.e go run . -e 'a' < toGrep.txt -> `< toGrep.txt` does not count so the len of flag.Args is still 0. But how do I really check if there are no files specified?

Reply:
No, you don't **specify a file** with <.

Reply:
That means if len(flag.Args()) == 0 -> i have no files specified but my program can still run because there is the case with "<"

Reply:
Exactly.